File: main/hw2/python/session-state-2.py
#!/usr/bin/env python3
import os
import json
import http.cookies

# 1. RETRIEVE SESSION
cookie = http.cookies.SimpleCookie(os.environ.get("HTTP_COOKIE"))
session_id = None
name = None

if "CGISESSID" in cookie:
    session_id = cookie["CGISESSID"].value
    session_file = f"/tmp/cgisess_{session_id}.json"
    
    if os.path.exists(session_file):
        with open(session_file, 'r') as f:
            data = json.load(f)
            name = data.get("username")

# 2. PRINT OUTPUT
print("Cache-Control: no-cache")
print("Content-type: text/html\n")

print("<html><head><title>Python Session Page 2</title></head><body>")
print("<h1>Python State Page 2</h1>")

if name:
    print(f"<p><b>Name:</b> {name}</p>")
else:
    print("<p><b>Name:</b> You do not have a name set (or session expired).</p>")

print("<br/><a href='session-state.py'>Back to Page 1</a><br/>")
print("<a href='/index.html'>Home</a>")
print("</body></html>")

File: main/hw2/ruby/session-state-2.rb
#!/usr/bin/ruby
require 'cgi'
require 'json'

cgi = CGI.new
session_id = cgi.cookies["CGISESSID"].first
name = nil

if session_id
  session_file = "/tmp/cgisess_#{session_id}.json"
  if File.exist?(session_file)
    data = JSON.parse(File.read(session_file))
    name = data['username']
  end
end

puts "Cache-Control: no-cache"
puts "Content-type: text/html\n\n"

puts "<html><head><title>Ruby Session Page 2</title></head><body>"
puts "<h1>Ruby State Page 2</h1>"

if name
  puts "<p><b>Name:</b> #{name}</p>"
else
  puts "<p><b>Name:</b> Session not found or no name set.</p>"
end

puts "<br/><a href='session-state.rb'>Back to Page 1</a><br/>"
puts "<a href='/index.html'>Home</a></body></html>"
puts "<form action='session-destroy.rb' method='get'><button>Destroy Session</button></form>"
